### PR TITLE
DM-37315: Remove psf_dimensions config field

### PR DESCRIPTION
**Pull request by arunkannawadi**

### CODE REVIEW

File: python/lsst/cell_coadds/_cellCoaddBuilder.py [hunk 1]
```diff
@@ -323,10 +318,7 @@ def run(  # type: ignore[override] # TODO: Remove after DM-34696 is fixed.
             outer_cell_size=cellInfo.outer_bbox.getDimensions(),
             inner_bbox=None,
             common=common,
-            psf_image_size=lsst.geom.Extent2I(
-                self.config.psf_dimensions,
-                self.config.psf_dimensions,
-            ),
+            psf_image_size=cellCoadds[0].psf_image.getDimensions(),
```
Comment (TallJimbo) [inline python/lsst/cell_coadds/_cellCoaddBuilder.py:321]: Do we have a guarantee that this will actually be the same for all cells?

**Review by TallJimbo — APPROVED** [review]

File: python/lsst/cell_coadds/_cellCoaddBuilder.py (hunk 1 above)
Comment (arunkannawadi) [reply python/lsst/cell_coadds/_cellCoaddBuilder.py:321]: Good point. As a config field in the single cell builder, they are supposed to be the same for all the cells. In the past, we've had something like round off errors that modifies the size by a pixel or two. I believe those were fixed, but our expectation is that they'd be the same for all cells.

Comment (TallJimbo) [reply python/lsst/cell_coadds/_cellCoaddBuilder.py:321]: If we're not sure they'll be the same, we may want to leave the config option and add logic to force all of the PSF size dimensions to the given one via padding or clipping.

Comment (arunkannawadi) [reply python/lsst/cell_coadds/_cellCoaddBuilder.py:321]: If they are not the same, then it's a bug elsewhere in the pipeline. We are forcing them to be the same by specifying the `psf_dimensions` in the `SCCBuilder`. Having it as a config option for `MultipleCellCoaddBuilder` opens up to the possibility that they can be different from that of `SCCBuilder` and we have no way of knowing what the user intended.

Comment (TallJimbo) [reply python/lsst/cell_coadds/_cellCoaddBuilder.py:321]: Ah, ok, then I think the original change here is just fine.

**Pull request merged**